### PR TITLE
test: playwright e2e test setup

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
[WEB-1113]

Added the following E2E test cases. 
- logging in
- navigation smoke test
- redirect to correct link after login

These are very minimal for now. we can extend it later.

## Test Plan
- Check if test cases are valid (didnt add edge cases since this is more like vital check)
- Check if CI works
- Check if E2E test works on local. [How to run test locally](https://github.com/determined-ai/determined/blob/292f3f703e65ab8d13e8d279d3de9d995fcd6eea/webui/react/src/e2e/README.md)
- Check if test is not flaky

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[WEB-1113]: https://hpe-aiatscale.atlassian.net/browse/WEB-1113?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/e2e/auth.spec.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    await dev.setServerAddress();
+  });
+
+  test('Login and Logout', async ({ page }) => {
+    await page.goto('/');
+
+    await test.step('Login steps', async () => {
+      await page.getByPlaceholder('username').fill(USERNAME);
+      await page.getByPlaceholder('password').fill(PASSWORD);
+      await page.getByRole('button', { name: 'Sign In' }).click();
```
Comment (ashtonG) [inline webui/react/src/e2e/auth.spec.ts:19]: consider abstracting the login process to the fixture as a page object.

**Comment by ashtonG** [comment]
What's the process for troubleshooting tests locally, given that we're using a docker image to run the tests here?

**Comment by keita-determined** [comment]
> What's the process for troubleshooting tests locally, given that we're using a docker image to run the tests here?

thats [how to run test locally](https://github.com/determined-ai/determined/blob/292f3f703e65ab8d13e8d279d3de9d995fcd6eea/webui/react/src/e2e/README.md)

File: webui/react/src/e2e/auth.spec.ts (hunk 1 above)
Comment (keita-determined) [reply webui/react/src/e2e/auth.spec.ts]: fixed

**Comment by ashtonG** [comment]
ah -- must have missed that when going through the readme
